django-stubs/template/backends/base.pyi

```diff
@@ -9,8 +12,15 @@ class BaseEngine:
     def __init__(self, params: Mapping[str, Any]) -> None: ...
     @property
     def app_dirname(self) -> Optional[str]: ...
-    def from_string(self, template_code: str) -> Template: ...
-    def get_template(self, template_name: str) -> Optional[Template]: ...
+    def from_string(self, template_code: str) -> EngineTemplate: ...
+    def get_template(self, template_name: str) -> EngineTemplate: ...
     @property
     def template_dirs(self) -> Tuple[str]: ...
     def iter_template_filenames(self, template_name: str) -> Iterator[str]: ...
+
+class EngineTemplate(Protocol):
```

Comment:
What about other props?
Like `template`, `origin`, and `backend`?

Examples:
- https://github.com/django/django/blob/de74a74b4b889c986cabab837bb03e2113880afe/django/template/backends/django.py#L48-L64
- https://github.com/django/django/blob/de74a74b4b889c986cabab837bb03e2113880afe/django/template/backends/jinja2.py#L55

Reply:
As well as the Django and Jinja classes, I think it could also be `dummy.Template` which only has `render()`?

https://github.com/django/django/blob/6f63e0ce8e36fbdb2d0c4dfd77ed3cf02e467f49/django/template/backends/dummy.py#L43

Reply:
Good point! 👍

Reply:
```suggestion
class _EngineTemplate(Protocol):
```